libraries/shared/src/RingBufferHistory.h

```diff
@@ -53,6 +53,18 @@ class RingBufferHistory {
             _numEntries++;
         }
     }
+    
+    // std::unique_ptr need to be passed as an rvalue ref and moved into the vector
+    void insert(T&& entry) {
+        // increment newest entry index cyclically
+        _newestEntryAtIndex = (_newestEntryAtIndex == _size - 1) ? 0 : _newestEntryAtIndex + 1;
```

Comment:
Is there a reason not to do `_newestEntryAtIndex = (_newestEntryAtIndex + 1) % _size`? Might just be me, but that's more recognizable, and it avoids the branch.